src/synth/fluid_synth.c

```diff
@@ -168,6 +173,15 @@ void fluid_synth_settings(fluid_settings_t* settings)
   fluid_settings_register_str(settings, "synth.default-soundfont", DEFAULT_SOUNDFONT, 0);
 #endif
 
+  /* basic channel settings for poly/mono mode */
+  /* synth.basic-channel: default: 0, min:0, max:255, no hint,no callback function */
+  fluid_settings_register_int(settings, "synth.basic-channel", 0, 0, 255, 0);
+  /* synth.basic-channel.mode: default: 0, min:0, max:3, no hint,no callback */
+  fluid_settings_register_int(settings, "synth.basic-channel-mode", 0, 0, 3, 0);
+  /* synth.basic-channel.modeval: default: 0, min:0, max:255, no hint,no callback*/
+  fluid_settings_register_int(settings, "synth.basic-channel-modeval", 0, 0, 255, 0);
```

Comment:
Initially these `synth.basic-channel*` settings was intended for the fluidsynth application console (using the -o option). But now,since the presence of commands `resetbasicchannels and setbasicchannels` theses commands can be put in a config file (using -f option).

Similarly since the presence of `fluid_synth_set_basic_channel() and fluid_synth_set_basic_channel()` API, an application can do what it need after synth creation. 
i think these settings could be removed. It will be done.